kstreams/streams.py

```diff
@@ -236,9 +240,10 @@ async def func_wrapper(self, func: typing.Awaitable) -> None:
             logger.exception(f"CRASHED Stream!!! Task {self._consumer_task} \n\n {e}")
 
     async def func_wrapper_with_typing(self) -> None:
-        while True:
+        while self.running:
             cr = await self.getone()
```

Comment:
I'm wondering if this doesn't still open the opportunity for an (async) race condition, where `getone` is still able to retrieve a record from Kafka, but then gets stuck on `async with self._is_processing`, as the lock is acquired by `stream.stop`

Reply:
Ah I read your comment just now, i think i'm describing Scenario 2?

Reply:
Yes.`Deserializer` will be deprecated and some point (the way to do it is with `middlewares`). If `deserializers` are not being used then we do not have to redefine the `getone` and we will avoid the race condition.